### PR TITLE
Feature: Add getProtocolVersion function

### PR DESCRIPTION
**Pull request by josemarinas**
## Description

Add `getProtopcolVersion` function

Task ID: [OS-629](https://aragonassociation.atlassian.net/browse/OS-629)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them when possible.
- [ ] I have updated the `CHANGELOG.md` file in the root folder of the package after the `[UPCOMING]` title and before the latest version.
- [ ] I have tested my code on the test network.

[OS-629]: https://aragonassociation.atlassian.net/browse/OS-629?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/client/src/internal/client/methods.ts [hunk 1]
```diff
@@ -866,4 +867,14 @@ export class ClientMethods extends ClientCore implements IClientMethods {
     }
     return toPluginRepo(pluginRepo, releaseMetadata, buildMetadata);
   }
+  public async getProtocolVersion(
```
Comment (heueristik) [inline modules/client/src/internal/client/methods.ts:938]: I imagined this to be a general function that you can apply to all OSx contracts (as mentioned here: https://discord.com/channels/672466989217873929/1132983258347487315/1132983261115723857).

We added the `protocolVersion` number function in v1.3.0.
This means that all prior contracts must be `1.0.0`.

**Review by heueristik — CHANGES_REQUESTED** [review]
We should provide a generic function and maybe a specifc one for the DAO

File: modules/client/src/internal/client/methods.ts [hunk 2]
```diff
@@ -866,4 +867,14 @@ export class ClientMethods extends ClientCore implements IClientMethods {
...
+    daoAddress: string,
+  ): Promise<[number, number, number]> {
+    if (!isAddress(daoAddress)) {
+      throw new InvalidAddressError();
+    }
+    const signer = this.web3.getConnectedSigner();
+    const daoInstance = DAO__factory.connect(daoAddress, signer);
```
Comment (heueristik) [inline modules/client/src/internal/client/methods.ts]: Use the `IProtocolVersion__factory` here

File: modules/client/src/internal/client/methods.ts [hunk 3]
```diff
@@ -925,4 +927,33 @@ export class ClientMethods extends ClientCore implements IClientMethods {
...
+    const signer = this.web3.getConnectedSigner();
+    const protocolInstance = IProtocolVersion__factory.connect(
+      contractAddress,
+      signer,
+    );
+    let version: [number, number, number];
+    try {
+      version = await protocolInstance.protocolVersion();
+    } catch (e) {
+      // ethers5 throws an cannot estimate gas limit which could mean a lot of things
```
Comment (heueristik) [inline modules/client/src/internal/client/methods.ts]: Hmm, this is not optimal indeed

File: modules/client/test/integration/client/methods.test.ts [hunk 4]
```diff
@@ -1313,6 +1316,24 @@ describe("Client", () => {
...
+      it("Should get the protocol version of a dao", async () => {
+        const ctx = new Context(contextParamsLocalChain);
+        const client = new Client(ctx);
+        const protocolVersion = await client.methods.getProtocolVersion(
+          daoAddress,
+        );
+        
+        expect(protocolVersion).toMatchObject([1, 3, 0]);
+      });
+
```
Comment (heueristik) [inline modules/client/test/integration/client/methods.test.ts:1354]: We should add a test where `[1,0,0]` is returned.

File: modules/client/test/helpers/deploy-v1-contracts.ts [hunk 5]
```diff
@@ -0,0 +1,563 @@
...
+      data: pluginItem.data,
+    },
+  ];
+  return createDAO(
+    deployment.daoFactory,
+    dummyMetadata,
+    pluginInstallItems,
+  );
+
+  // try {
```
Comment (heueristik) [inline modules/client/test/helpers/deploy-v1-contracts.ts]: Why this is commented out?

**Review by heueristik — DISMISSED** [review]

**Review by mathewmeconry — DISMISSED** [review]

**Review by heueristik — APPROVED** [review]

**Pull request merged**